### ISSUE TITLE
[Bug]  scoop uninstall scoop - He deleted something that was not within his competence.

### ISSUE THREAD

**Issue by alexeyp0708** [ScoopInstaller/Scoop#5726]
According to this issue https://github.com/ScoopInstaller/Scoop/issues/249
I changed the installation location via `[environment]::setEnvironmentVariable('SCOOP','Z:\usr','User')`
Next I installed SCOOP
`Set-ExecutionPolicy RemoteSigned -Scope CurrentUser`
`irm get.scoop.sh | iex`
After installation, I realized that the installer for SCOOP does not create a separate folder and created all the working folders in the current directory (Z:\usr)
I decided to remove SCOOP
`scoop uninstall scoop`
As a result, all files and folders in `Z:\usr` began to be deleted.
He deleting until he threw an error on the blocked file.
Now I don’t know what exactly was deleted from me.
I don't know how to restore files now.
I'm shocked. I hoped that it would delete safely without affecting files in the current directory and would delete exactly those folders that it had installed.
there were no warnings that the deletion would also affect third-party files

Complete shit.

**Comment by alexeyp0708** [comment]
Восcтановил файлы через r-studio.
But this behavior is walking a fine line.
Please correct and remove the component safely. (create a registry of component files during installation, and delete only them - if the folder is not empty, then complete the removal and notify about it)

**Comment by HUMORCE** [comment]
Losing files can be frustrating, but this is a human error, not a bug.

Scoop considers `Z:\usr` as its root directory and 'competence scope', when you set the environment variable.

Although the guide in #249 was published in 2014, but it still applies to the current version of Scoop, and it contains a backup step.

Related:
- https://github.com/ScoopInstaller/Scoop?tab=readme-ov-file#installation
- https://github.com/ScoopInstaller/Install?tab=readme-ov-file#advanced-installation

**Comment by alexeyp0708** [comment]
@HUMORCE 
https://github.com/ScoopInstaller/Scoop/blob/f93028001fbe5c78cc41f59e3814d2ac8e595724/bin/uninstall.ps1#L57
If you think that forced deletion of a directory, without regard to the human factor, is good, then planes will crash more often with such programmers.

**Comment by alexeyp0708** [comment]
Also, this behavior is a tasty morsel for malware,
Just change the SCOOP environment variable,
and subsequently by running "uninstall.ps1" another directory with all its contents will be deleted.

**Comment by HUMORCE** [comment]
> Also, this behavior is a tasty morsel for malware,

Malicious can do better without this in fact.

> Or another option - when installing in a non-empty directory, throw an error and stop the installation.

That's good and enough. tracking files is not a good idea.

---

However, I would still identify the entire process of your missing files as Human error. you are installing Scoop with the advanced/unrecommended way in first time installation.

**Comment by HUMORCE** [comment]
The suggestion for non-empty directory check was added to the https://github.com/ScoopInstaller/Install/issues/60

For consistency, we should prompting more information when executing `scoop uninstall scoop`.

**Comment by alexeyp0708** [comment]
> However, I would still identify the entire process of your missing files as Human error. you are installing Scoop with the advanced/unrecommended way in first time installation.
@HUMORCE  
I agree with you that this is a human error. But human errors are caused by the imperfection of the system. The system must exclude such fatal errors.
In my case, my entire development environment is on a separate disk and is not tied to a specific computer.
In your installation, to change the installation path, you have to change the SCOOP variable.
Personally, I thought that when installing, your program takes care of the fact that it will create its own directory. And the SCOOP variable is needed to indicate the parent directory. For example, like in Windows - ProgramFiles directory or User/AppData directory.
But everything didn't go according to plan.
I still have to sort out this problem. Since this script affected many directories.

When installing, you can simply check for an empty directory. if the directory is empty, then install it into it, otherwise create your own Scoope directory. Or, as a last resort, ask the user to set the SCOOPE variable to an empty directory.

In this commentary, I expressed additional information and opinion that may help eliminate human error in the future.

Thank you for your attention.

**Comment by HUMORCE** [comment]
> the SCOOP variable is needed to indicate the parent directory. For example, like in Windows - ProgramFiles directory or User/AppData directory.

`$env:SCOOP` for set the Scoop Root Direcotry, like:

```
⭕️ $JAVA_HOME = ~/java/openjdk11
❌ $JAVA_HOME = ~/java

⭕️ $GOPATH = ~/go
❌ $GOPATH = ~

etc.
```
That's why I think it's a human error. I have not meet a situation where an environment variable is pointed to a parent directory, which may appear in the installation wizard of some programs:

> Selected: c/programs/ , then actual install to: c/programs/app

**Comment by alexeyp0708** [comment]
> $env:SCOOP for set the Scoop Root Direcotry, like:
Please indicate where this is in the documentation where your environment variables are described.

You must understand that everyone has their own intuition and experience in using programs. The program should not cause harm even if a human error occurs.ё

**Comment by HUMORCE** [comment]
> Please indicate where this is in the documentation where your environment variables are described.

If you follow the installation guide, you will see the name of the parameter as `-ScoopDir`: https://github.com/ScoopInstaller/Install?tab=readme-ov-file#advanced-installation

> You must understand that everyone has their own intuition and experience in using programs. The program should not cause harm even if a human error occurs.ё

Of course, I know and I could not believe the mistake was made by you when I noticed you were using *nix-like directory structure in Windows.

---

There's no point in arguing this now.  

Your issue has been tracked, once it is improved,**THIS** issue will not occur again. But what about the others? We must be vigilant when trying new things.

**Comment by alexeyp0708** [comment]
> Please indicate where this is in the documentation where your environment variables are described.

I'm sorry. I found it in the description on the README.md page in [ScoopInstaller/Install](https://github.com/ScoopInstaller/Install).

The installation information is divided into different parts, so I was not able to read it in time.
I needed to install the required package according to someone else's instructions, but in a specific directory. I didn’t care about additional one-time information at that time.
